x-pack/plugin/core/src/internalClusterTest/java/org/elasticsearch/xpack/core/rest/action/XPackUsageRestCancellationIT.java

```diff
@@ -90,6 +91,7 @@ public void testCancellation() throws Exception {
         assertThat(future.isDone(), equalTo(false));
         awaitTaskWithPrefix(actionName);
 
+        blockingXPackUsageActionExecuting.await();
```

Comment:
Does it make sense to wait with some timeout just to safeguard that the test doesn't stuck forever?

Reply:
our test suites have a timeout for such problems https://github.com/elastic/elasticsearch/blob/a8258bde1643f8cf5eda0ee437e7c2458f62fada/test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java#L160